homeassistant/components/skybell/binary_sensor.py

```diff
@@ -54,9 +73,9 @@ def __init__(self, device, sensor_type):
         super().__init__(device)
         self._sensor_type = sensor_type
```

Comment:
`sensor_type` is only used to access the `SENSOR_TYPES` dictionary. If we instead just pass the metadata in `__init__`, we don't need it anymore and can further simplify the code.

Reply:
**Update**: Saw that `self._sensor_type` is used elsewhere. So that won't entirely work. Maybe it's just an idea to do something like this instead? Although I'm not sure how useful it would really be (for this integration).
```py
metadata = SENSOR_TYPES[self._sensor_type]
```

Reply:
Not really, it is also used in the schema definition, line 50, so I am afraid we are stuck with SENSOR_TYPES.

Reply:
I'm currently checking other integrations, that's where the idea came from.
Take a look at this
https://github.com/home-assistant/core/blob/0803b2aecd537b836338f4ff075afff242deceba/homeassistant/components/nut/sensor.py#L104-L107

I think here it could be useful to add
```py
metadata = SENSOR_TYPES[self._type]
```
so it can be replaced for all those calls. That should make it easier to read. In that spirit the suggestion here would more be for consistency.

Reply:
Good idea, exact I added self._meta, so that I could replace all SENSOR_TYPES[self._sensor_type] (it was used in a method as well).